### PR TITLE
Add support for seconds to the timepicker

### PR DESCRIPTION
**Pull request by chriscareycode**
As the title says, this PR is to add support for seconds to the time picker. When using a customTimeInput it will pass HH:mm:ss to the component with this change

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Awesome, can you add some tests to cover the functionality and a docs site example so that others know how to use this too?

**Comment by chriscareycode** [comment]
Yes I will add some test, docs, and a prop to enable. Thanks Martijn

**Comment by chriscareycode** [comment]
@martijnrusschen  I can run the test suite locally with 'yarn run test' but not sure that it's possible to run this codecov locally to verify I've made progress except for actually checking in to the PR. Any tips there?

**Comment by chriscareycode** [comment]
I have added test, docs, and a prop to enable this feature

**Comment by martijnrusschen** [comment]
Let's revisit this PR

**Comment by chriscareycode** [comment]
Will take a look at these test errors and see what I can do

**Comment by AquApp** [comment]
Hello 
Is it possible to merge or there are still things to do?
Thanks

**Comment by martijnrusschen** [comment]
Can you pull in the latest changes from master? It doesn't look like the CI picked up this PR yet.

**Comment by AquApp** [comment]
Sorry i haven't right to modify this

**Comment by volkan-anchor** [comment]
Any updates on this @martijnrusschen @AquApp ?

**Comment by chriscareycode** [comment]
This PR has a number of commits and merges. I'd like to flatten it into a single commit to clean it up and make it easier to rebase up to master. Have been swamped the past couple weeks at work but hope to get to it soon.

**Comment by volkan-anchor** [comment]
Thanks Chris(@chriscareycode) looking forward to it!

**Comment by jennyk1** [comment]
posted a workaround for this issue here: https://github.com/Hacker0x01/react-datepicker/issues/3757

**Comment by chriscareycode** [comment]
Squashed the multiple commits on this branch into a single commit and force pushed. We now have one commit rebased on top of master. This should make a cleaner PR and easier to move forward.

File: test/time_input_test.js [hunk 1]
```diff
@@ -40,6 +40,15 @@ describe("timeInput", () => {
     expect(timeComponent.state("time")).to.equal("13:00");
   });
 
+  xit("should trigger onChange event (with seconds)", () => {
```
Comment (martijnrusschen) [inline test/time_input_test.js:43]:
```suggestion
  it("should trigger onChange event (with seconds)", () => {
```

Comment (martijnrusschen) [reply test/time_input_test.js:43]: Please try to make this test work.

Comment (chriscareycode) [reply test/time_input_test.js:43]: Hi Martijn. Will do. I have this fixed in local dev. Just need to get it cleaned up and add another test. I figured out why the other tests are not working, like the one above this one. When input.simulate("change") fires, it is changing the state, but then the getDerivedStateFromProps() function inside inputTime is firing and overwriting the change with the prop. Hoping to get a pattern here you can use to re-enable some of these other tests that are disabled in this file.

**Comment by ouedyan** [comment]
Any update on this @chriscareycode ?

**Pull request closed without merging**